lms/djangoapps/courseware/middleware.py

```diff
@@ -17,3 +18,20 @@ def process_exception(self, _request, exception):
         """
         if isinstance(exception, Redirect):
             return redirect(exception.url)
+
+
+class CacheCourseIdMiddleware(object):
```

Comment:
@waheedahmed  not sure about this approach, will this add `course_id` in all requests or specific only to contact page requests.

Reply:
@tasawernawaz I wasn't sure about this either but I couldn't find other way to do this. I'll look into it more if I can find another way.

Reply:
@staubina We may need your opinion on this approach.